include/emp/bits/BitVector.hpp

```diff
@@ -701,6 +700,72 @@ namespace emp {
       return (int) (pos_found + (field_id * FIELD_BITS));
     }
 
+    /// Pop the last bit in the vector
+    bool PopBack() {
+      bool val = Get(num_bits-1);
+      if (val)  Set(num_bits-1, 0);
+      Resize(num_bits - 1);
+      return val;
+    }
+
+    /// Push given bit(s) onto the vector
+    void PushBack(bool bit=true, size_t num=1) {
+      Resize(num_bits + num);
+      for (size_t i=0; i < num; i++)
+        Set(num_bits-1, bit);
+    }
+
+    // Returns mask that keeps bits higher than index
+    void Mask_High(size_t index) {
+      size_t field_id = FieldID(index);
+      int field_t_bits = sizeof(bit_set[0]) * 8;
+      for (size_t i = 0; i < field_id; i++)
+        bit_set[i] = 0;
+      if (index >= 0)
+        bit_set[field_id] &= (MaskLow<field_t>((num_bits - index - 1) % field_t_bits) << (index + 1));
+    }
+
+    // Returns mask that keeps bits lower than index
+    void Mask_Low(size_t index) {
+      size_t field_id = FieldID(index);
+      int field_t_bits = sizeof(bit_set[0]) * 8;
```

Comment:
use this value instead of recalculating https://github.com/devosoft/Empirical/blob/62fccad1c411c3cbae4a64f9387dee01d1a66648/include/emp/bits/BitVector.hpp#L57